dor/adapters/catalog.py

```diff
@@ -85,6 +97,22 @@ def get_by_alternate_identifier(self, identifier: str) -> models.Revision | None
                 latest_revision = revision
         return latest_revision
 
+    def find_by_file_set(self, requested_identifier: str) -> list[models.Revision]:
+        revisions = []
+        file_set_identifier = uuid.UUID(requested_identifier)
+        referenced_file_set_identifier = AlternateIdentifier(
+            type="function:version",
```

Comment:
We discussed offline the name. In my view we don't have to use the `function:` language here and integrate with our `UseFunction` CV. Maybe a new one like `AlternateIdentifierType` that includes "DLXS" might be appropriate. I like something like "Source" or "Original".

I also floated the idea of a separate nullable field like `source_identifier` instead of using `AlternateIdentifier`. Not sure if that's better, but thought it worth mentioning.

Reply:
If we taking a typical ORM-ish approach, `source_identifier` might make sense. Since we're taking a document-database approach, we don't have to transform the data into another format.

Reply:
I added `function:copy-of` to `UseFunctions` because in my mind we're building a vocabulary of functions.